pkg/resourcecollector/resourcecollector.go

```diff
@@ -0,0 +1,356 @@
+	}
+	r.dynamicInterface, err = dynamic.NewForConfig(config)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func resourceToBeCollected(resource metav1.APIResource) bool {
+	// Deployment is present in "apps" and "extensions" group, so ignore
+	// "extensions"
+	if resource.Group == "extensions" && resource.Kind == "Deployment" {
+		return false
+	}
+
+	switch resource.Kind {
+	case "PersistentVolumeClaim",
+		"PersistentVolume",
+		"Deployment",
+		"StatefulSet",
+		"ConfigMap",
+		"Service",
+		"Secret",
+		"DaemonSet",
+		"ServiceAccount",
+		"ClusterRole",
+		"ClusterRoleBinding":
+		return true
+	default:
+		return false
+	}
+}
+
+// GetResources gets all the resources in the given list of namespaces which match the labelSelectors
+func (r *ResourceCollector) GetResources(namespaces []string, labelSelectors map[string]string) ([]runtime.Unstructured, error) {
+	err := r.discoveryHelper.Refresh()
+	if err != nil {
+		return nil, err
+	}
+	allObjects := make([]runtime.Unstructured, 0)
+
+	for _, group := range r.discoveryHelper.Resources() {
+		groupVersion, err := schema.ParseGroupVersion(group.GroupVersion)
+		if err != nil {
+			return nil, err
+		}
+		if groupVersion.Group == "extensions" {
+			continue
+		}
+
+		// Map to prevent collection of duplicate objects
+		resourceMap := make(map[types.UID]bool)
+		for _, resource := range group.APIResources {
+			if !resourceToBeCollected(resource) {
+				continue
+			}
+
+			for _, ns := range namespaces {
+				var dynamicClient dynamic.ResourceInterface
+				if !resource.Namespaced {
+					dynamicClient = r.dynamicInterface.Resource(groupVersion.WithResource(resource.Name))
+				} else {
+					dynamicClient = r.dynamicInterface.Resource(groupVersion.WithResource(resource.Name)).Namespace(ns)
+				}
+
+				var selectors string
+				// PVs don't get the labels from their PVCs, so don't use
+				// the label selector
+				if resource.Kind != "PersistentVolume" {
+					selectors = labels.Set(labelSelectors).String()
+				}
+				objectsList, err := dynamicClient.List(metav1.ListOptions{
+					LabelSelector: selectors,
+				})
+				if err != nil {
+					return nil, err
+				}
+				objects, err := meta.ExtractList(objectsList)
+				if err != nil {
+					return nil, err
+				}
+				for _, o := range objects {
+					runtimeObject, ok := o.(runtime.Unstructured)
+					if !ok {
+						return nil, fmt.Errorf("Error casting object: %v", o)
+					}
+
+					collect, err := r.objectToBeCollected(labelSelectors, resourceMap, runtimeObject, ns)
+					if err != nil {
+						return nil, fmt.Errorf("Error processing object %v: %v", runtimeObject, err)
+					}
+					if !collect {
+						continue
+					}
+					metadata, err := meta.Accessor(runtimeObject)
+					if err != nil {
+						return nil, err
+					}
+					allObjects = append(allObjects, runtimeObject)
+					resourceMap[metadata.GetUID()] = true
+				}
+			}
+		}
+	}
+
+	return allObjects, nil
+}
+
+func (r *ResourceCollector) objectToBeCollected(
```

Comment:
A comment on top, as to what this functions purpose is.

Reply:
Done